File: 004/004.rb
#!/usr/local/bin/ruby -w

# Find the largest palindrome made from the product of two 3-digit numbers.

def palindrome?(k)
    (0..k.length / 2).each do |s|
        return false if k[s] != k[-(1 + s)]
    end
    return true
end

control = 0
(100..999).to_a.reverse.each do |i|
    (100..999).to_a.reverse.each do |j|
        k = i * j
        if palindrome?(k.to_s) == true
            control = k if k > control
        end
    end
end

puts("The largetst palindrome in question is #{control}")

File: 004/004.java
// Find the largest palindrome made from the product of two 3-digit numbers.

class Main {
    public static boolean isPalindrome(int k) {
        String r = String.valueOf(k);
        for (int s = 0; s < r.length() / 2; s++) {
            if (r.charAt(s) != r.charAt(r.length() - (1 + s))) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        int control = 0;
        for (int i = 999; i > 99; i--) {
            for (int j = 999; j > 99; j--) {
                int  k = i * j;
                if (isPalindrome(k) == true) {
                    if (k > control) {
                        control = k;
                    }
                }
            }
        }
        System.out.printf("The largest palindrome in question is %d.\n", control);
    }
}
